storage/onpremise/mongodb/main.tf

```
@@ -47,17 +58,47 @@ resource "helm_release" "mongodb" {
           }
         ] : []
       } : {}
+
       # As the parameter 'tls.mTLS.enabled' set to false doesn't seem to work (chart v15.1.4) this an
       # alternative to allow mTLS to not be mandatory
       "extraFlags" = "--tlsAllowConnectionsWithoutCertificates"
 
-      "persistentVolumeClaimRetentionPolicy" = {
+      "persistentVolumeClaimRetentionPolicy" = var.persistent_volume != null ? {
         "enabled"     = "true"
-        "whenDeleted" = "Delete"
-      }
+        "whenDeleted" = var.persistent_volume.reclaim_policy
+      } : {}
     })
   ]
 
+  dynamic "set" {
+    for_each = var.persistent_volume == null ? [1] : []
+    content {
+      name  = "persistence.enabled"
+      value = false
+    }
+  }
+  dynamic "set" {
+    for_each = var.persistent_volume != null && try(var.persistent_volume.storage_provisioner, "") != "" ? [1] : []
+    content {
+      name  = "persistence.storageClass"
+      value = kubernetes_storage_class.mongodb[0].metadata[0].name
+    }
+  }
+  dynamic "set" {
+    for_each = var.persistent_volume != null ? [1] : []
+    content {
+      name  = "persistence.accessMode[0]"
+      value = var.persistent_volume.access_mode[0]
+    }
+  }
+  dynamic "set" {
+    for_each = var.persistent_volume != null && try(var.persistent_volume.resources.requests.storage, "8Gi") != "8Gi" ? [1] : []
```

Why do you compare to 8Gi?

```suggestion
    for_each = try(var.persistent_volume.resources.requests.storage, "8Gi") != "8Gi" ? [1] : []
```

---

It's the Helm chart default value. but a can/coalesce structure is indeed better